### PR TITLE
Copy ParentTraceContext when cloning ExecutionStartedEvent

### PR DESCRIPTION
**Pull request by bachuv**
This PR adds a line to copy `ExecutionStartedEvent.ParentTraceContext` when cloning `ExecutionStartedEvent` in `CreationRequestReceived.Clone()`. This change is necessary for Distributed Tracing V2 to work.

I created this PR as a draft because a new version of DT.Core needs to be released with a bug fix for this to work so this PR shouldn't be merged until then.

### CODE REVIEW

**Review by sebastianburckhardt — APPROVED** [review]
LGTM, thanks!

File: src/DurableTask.Netherite/Events/PartitionEvents/External/FromClients/CreationRequestReceived.cs [hunk 1]
```diff
@@ -75,6 +75,7 @@ public override PartitionEvent Clone()
                     ExtensionData = ee.ExtensionData,
                     OrchestrationInstance = ee.OrchestrationInstance,
                     Timestamp = this.CreationTimestamp,
+                    ParentTraceContext = ee.ParentTraceContext,
                 },
                 SequenceNumber = tm.SequenceNumber,
                 OrchestrationInstance = tm.OrchestrationInstance,
```
Comment (davidmrdavid) [inline src/DurableTask.Netherite/Events/PartitionEvents/External/FromClients/CreationRequestReceived.cs:81]: non blocking question: it appears this copy of the execution started event is very manual, so I worry that we'll break it again in the future if we add a new field. Is there a more future-proof way of doing this copy? For example, via [Object.MemberwiseClone()](https://learn.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-7.0)?

If so, perhaps we can add a TODO comment on this line to object copy to consider changing the cloning approach

Comment (bachuv) [reply src/DurableTask.Netherite/Events/PartitionEvents/External/FromClients/CreationRequestReceived.cs:81]: @sebastianburckhardt, what are your thoughts on this?

**Comment by bachuv** [comment]
@sebastianburckhardt, can we merge this PR to be a part of the v1.4.1 release?

File: src/DurableTask.Netherite/Events/PartitionEvents/External/FromClients/CreationRequestReceived.cs (hunk 1 above)
Comment (sebastianburckhardt) [reply src/DurableTask.Netherite/Events/PartitionEvents/External/FromClients/CreationRequestReceived.cs:81]: Using `Object.MemberwiseClone` seems like a good idea to me.

**Comment by bachuv** [comment]
Leaving a note that this PR depends on DT.Core > 2.15.1.

**Review by sebastianburckhardt — APPROVED** [review]

**Pull request merged**